### ISSUE TITLE
Music file types (extensions?) list is not honored?

### ISSUE THREAD

**Issue by ohnonot** [airsonic/airsonic#1227]
## Problem description

I would like to exclude all lossless formats from getting indexed (added to the database) at all.
I'm trying to achieve this logged in as admin, Settings => Music Files is currently filled with  
`mp3 ogg oga aac m4a wma ape mpc shn mka opus` which I thought to mean that only files with this extension are considered (case insensitive).

### Steps to reproduce

With the settings as described, I do both "Scan Media Folders" and "Clean-up database". Then I open my Android app DSub, swipe down to get the latest database from the server.
Whatever I do, FLAC files are still being shown, and are playable too.

## System information

 * **Airsonic version**:
Still on 10.3.1-RELEASE – May 21, 2019 :blush: but this problem has been with me for a long time.

 * **Operating system**:
Debian oldstable (stretch) no-GUI server install.

 * **Java version**: 
Apache Tomcat/8.5.40, java 1.8.0_222

 * **Proxy server**:
nginx

 * **Client**:
Admin in Firefox, actual listening on Android client (DSub)

**Comment by eharris** [comment]
@ohnonot Please try the patch mentioned above and verify that it resolves your issue.

**Comment by ohnonot** [comment]
Thanks!
AFAIU, and according to [this](https://airsonic.github.io/docs/install/source/) I'd have to do a bit of JAVA compiling?
My server is on debian oldstable (stretch), and the **maven** version available is 3.3.9 - I hope that's recent enough.
I installed:
`apt install openjdk-8-jre openjdk-8-jdk maven`
I cloned https://github.com/eharris/airsonic, changed into the airsonic directory and issued
`git checkout filter-media-files`
and confirmed with
`git log`
that the referenced commit "Media file scan will now heed configured music/video file types" shows up.
Then
`mvn clean package`
It compiles, makes tests, all apparently without errors, but then it hangs on
```
Sep 14, 2019 10:55:39 PM org.apache.maven.wagon.providers.http.httpclient.impl.execchain.RetryExec execute
INFO: I/O exception (java.net.SocketException) caught when processing request to {s}->https://repo.maven.apache.org:443: Connection timed out (Read failed)

```
(repeats)
for maybe 90min already. I have to cut it now. Maybe tomorrow.

**Comment by jvoisin** [comment]
Look like a maven-repository failure. Can you try again?

**Comment by ohnonot** [comment]
I tried again, and it compiled.

I dropped the resulting .war into my existing airsonic directory, and it is working as desired; files with other extensions than listed are not picked up anymore.

Also on the Android client, everything seems to be working.

Of course I have a lot of empty directories now; it would be nice to add an option to not even display directories that do not contain anything playable.
